vcd/resource_vcd_vapp_access_control_test.go

```diff
@@ -0,0 +1,337 @@
+					}),
+					testAccFindValuesInSet(resourceAC3, "shared_with", map[string]string{
+						"subject_name": "ac-user3",
+						"access_level": types.ControlAccessReadOnly,
+					}),
+				),
+			},
+			resource.TestStep{
+				Config: updateText,
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckVcdVappAccessControlExists(resourceAC0, testConfig.VCD.Org, testConfig.VCD.Vdc),
+					resource.TestCheckResourceAttr(resourceAC0, "shared_with_everyone", "false"),
+					resource.TestCheckResourceAttr(resourceAC1, "shared_with.#", "1"),
+					testAccFindValuesInSet(resourceAC1, "shared_with", map[string]string{
+						"subject_name": "ac-user1",
+						"access_level": types.ControlAccessReadWrite,
+					}),
+					resource.TestCheckResourceAttr(resourceAC2, "shared_with.#", "2"),
+					testAccFindValuesInSet(resourceAC2, "shared_with", map[string]string{
+						"subject_name": "ac-user1",
+						"access_level": types.ControlAccessReadWrite,
+					}),
+					resource.TestCheckResourceAttr(resourceAC3, "shared_with.#", "3"),
+					testAccFindValuesInSet(resourceAC3, "shared_with", map[string]string{
+						"subject_name": "ac-user1",
+						"access_level": types.ControlAccessReadWrite,
+					}),
+				)},
+
+			// Tests import by name
+			resource.TestStep{
+				Config:            configText,
+				ResourceName:      "vcd_vapp_access_control.AC-Vapp1",
+				ImportState:       true,
+				ImportStateVerify: true,
+				ImportStateIdFunc: importStateIdOrgVdcObject(testConfig, "Vapp-AC-1"),
+			},
+			// Tests import by ID
+			resource.TestStep{
+				Config:            configText,
+				ResourceName:      "vcd_vapp_access_control.AC-Vapp2",
+				ImportState:       true,
+				ImportStateVerify: true,
+				ImportStateIdFunc: importStateIdViaResource("vcd_vapp_access_control.AC-Vapp2"),
+			},
+		},
+	})
+}
+
+func testAccCheckVcdVappAccessControlExists(resourceName string, orgName, vdcName string) resource.TestCheckFunc {
+	return func(s *terraform.State) error {
+
+		rs, ok := s.RootModule().Resources[resourceName]
+		if !ok {
+			return fmt.Errorf("not found: %s", resourceName)
+		}
+
+		if rs.Primary.ID == "" {
+			return fmt.Errorf("no vApp ID is set")
+		}
+
+		conn := testAccProvider.Meta().(*VCDClient)
+
+		org, err := conn.VCDClient.GetAdminOrgByName(orgName)
+		if err != nil {
+			return fmt.Errorf("error: could not find Org: %s", err)
+		}
+		vdc, err := org.GetVDCByName(vdcName, false)
+		if err != nil {
+			return fmt.Errorf("error: could not find VDC: %s", err)
+		}
+
+		vapp, err := vdc.GetVAppById(rs.Primary.ID, false)
+		if err != nil {
+			return fmt.Errorf("error: could not find vApp %s: %s", rs.Primary.ID, err)
+		}
+		_, err = vapp.GetAccessControl(tenantContext)
+		if err != nil {
+			return fmt.Errorf("error: could not get access control for vApp %s: %s", vapp.VApp.Name, err)
+		}
+
+		return nil
+	}
+}
+
+func testAccCheckVappAccessControlDestroy(orgName, vdcName string, vappNames []string) resource.TestCheckFunc {
```

Comment:
I hit an issue with destroy. Removed the resource:

```
An execution plan has been generated and is shown below.
Resource actions are indicated with the following symbols:
  - destroy

Terraform will perform the following actions:

  # vcd_vapp_access_control.demo_vapp_sharing will be destroyed
  - resource "vcd_vapp_access_control" "demo_vapp_sharing" {
      - everyone_access_level = "Change" -> null
      - id                    = "urn:vcloud:vapp:0983cdbb-61b2-460d-8e63-cdfeec8dbb85" -> null
      - shared_with_everyone  = true -> null
      - vapp_id               = "urn:vcloud:vapp:0983cdbb-61b2-460d-8e63-cdfeec8dbb85" -> null

      - shared_with {
          - access_level = "FullControl" -> null
          - user_id      = "urn:vcloud:user:ee04f71a-27a5-4519-865a-fb3ba35538e3" -> null
        }
    }

Plan: 0 to add, 0 to change, 1 to destroy.

Do you want to perform these actions?
  Terraform will perform the actions described above.
  Only 'yes' will be accepted to approve.

  Enter a value: yes

vcd_vapp_access_control.demo_vapp_sharing: Destroying... [id=urn:vcloud:vapp:0983cdbb-61b2-460d-8e63-cdfeec8dbb85]
vcd_vapp_access_control.demo_vapp_sharing: Still destroying... [id=urn:vcloud:vapp:0983cdbb-61b2-460d-8e63-cdfeec8dbb85, 10s elapsed]
vcd_vapp_access_control.demo_vapp_sharing: Destruction complete after 13s

Apply complete! Resources: 0 added, 0 changed, 1 destroyed.
```

But in GUI this is what I got:

<img width="1152" alt="Screenshot 2020-08-27 at 18 08 40" src="https://user-images.githubusercontent.com/11330689/91460530-8e62dd80-e890-11ea-851b-cfdce15113f4.png">

Reply:
It is not a bug.
The GUI doesn't have a way of saying "not sharing", so it shows that is "shared with", and no user or group is ticked.
It shows "Users (0)" and "Groups (0)"